sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -682,12 +682,29 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                     }
                     case "numeric_inc":
                     case "sign":
-                    case "log10":
-                    case "ln":
                     case "abs": {
                         return this.compilePolymorphicFunction(call, node, type,
                                 ops, 1);
                     }
+                    case "log10":
+                    case "ln":
+                    {
+                        // Cast to Double if the type isn't Decimal
+                        DBSPExpression arg = ops.get(0);
+                        if (!arg.type.is(DBSPTypeDecimal.class)) {
```

Comment:
It is required for `PostgresFloat8Tests#nonPostgresTests`. 

Error message if we don't cast: 

```rust
error[E0425]: cannot find function `ln_i32` in this scope
   --> src/lib.rs:822:23
    |
822 |             Tup1::new(ln_i32(1i32))
    |                       ^^^^^^ not found in this scope
```

Reply:
This sucks. I think this is another Calcite bug really.
But how about f32?

Reply:
Same issue. 

```rust
error[E0425]: cannot find function `ln_f` in this scope
   --> src/lib.rs:228:23
    |
228 |             Tup1::new(ln_f(F32::new(1.0)))
    |                       ^^^^ help: a function with a similar name exists: `ln_d`
```

Reply:
No, it's not the same issue. Here you are calling a function which you didn't implement.
You have to implement it.
That's what "polymorphic" means: the same SQL function has multiple implementations depending on the types of the arguments.

Reply:
Rust has no polymorphic functions, so we have to do this juggling.

Reply:
The question is whether Rust even supports ln on floats.
For is_nan I am pretty sure you have to have both.
But here if Rust does always convert to f64 you don't need a polymorphic function.

Reply:
Oh, I meant in terms that, according to the Postgres docs `ln` isn't defined for real. So I thought Calcite should've inserted casts to double. 

```sql
ln ( numeric ) → numeric
ln ( double precision ) → double precision
-- Natural logarithm
ln(2.0) → 0.6931471805599453
```

But yeah, we can implement it for real too.

Reply:
It may be that the FP unit always computes on (the C language) double, and if you want float it casts back and forth. No point in implementing it on float it it doesn't really exist on float.
Let's keep it simple for now, just doubles.
So keep the cast in the compiler.
But is_inf needs to be polymorphic.